src/main/scala/akka/stream/contrib/ZipInputStreamSource.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import java.util.zip.{ ZipEntry, ZipInputStream }
+import akka.stream.Attributes.{ InputBuffer, name }
+import akka.stream.contrib.ZipInputStreamSource.{ DefaultChunkSize, ZipEntryData }
+import akka.stream.impl.Stages.DefaultAttributes.IODispatcher
+import akka.stream.scaladsl.Source
+import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, OutHandler }
+import akka.stream.{ Attributes, Outlet, SourceShape }
+import akka.util.ByteString
+import akka.util.ByteString.ByteString1C
+import scala.annotation.tailrec
+import scala.concurrent.{ Future, Promise }
+
+/**
+ * This companion defines a factory for [[ZipInputStreamSource]] instances,
+ * see [[ZipInputStreamSource.apply]].
+ */
+object ZipInputStreamSource {
+
+  val DefaultChunkSize: Int = 8192
+
+  /**
+   * Data type for zip entries.
+   *
+   * @param name file name
+   * @param creationTime The last modification time of the entry in milliseconds
+   *          since the epoch, or -1 if not specified
+   */
+  case class ZipEntryData(name: String, creationTime: Long)
+
+  /**
+   * Factory for [[ZipInputStreamSource]] instances wrapped
+   * into [[Source]].
+   *
+   * @param in a function that builds a [[ZipInputStream]]
+   * @param chunkSize the size of the chunks
+   * @return [[ZipInputStreamSource]] instance
+   */
+  def apply(
```

Comment:
I think it's the consequence of having `alignParameters` set to `true` in the `formattingPreferences`. Just to clarify, would you want to change this?.